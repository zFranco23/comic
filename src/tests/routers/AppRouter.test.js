import { mount} from "enzyme"
import { AuthContext } from "../../auth/AuthContext";
import AppRouter from "../../routers/AppRouter"



describe('Pruebas en <AppRouter />', () => {
    
    const contextValue={
        dispatch : jest.fn(),
        user:{
            logged:false
        }
    }
    test('Debe mostrar el login si no esta autenticado', () => {
        
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect(wrapper).toMatchSnapshot();
    })
    
    test('Debe mostrar el componente marvel si estÃ¡ autenticado', () => {
        const contextValue={
            dispatch : jest.fn(),
            user:{
                name:"Franco",
                id :"zFranco23",
                logged:true,
            }
        }
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        )

        expect(wrapper.find('.navbar').exists()).toBe(true);
    })
    
})
