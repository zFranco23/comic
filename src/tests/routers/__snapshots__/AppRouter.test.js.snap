// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> Debe mostrar el login si no esta autenticado 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login
    </h1>
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
